### ISSUE TITLE
No Applicable Method from LLVM back-end

### ISSUE THREAD

**Issue by waywardmonkeys** [dylan-lang/opendylan#1017]
When building the ``objective-c`` test suite (https://github.com/dylan-foundry/objc-dylan/) with the LLVM back-end, this ends up happening:

```
Generating code for library objc-test-suite
computation return t30: Error:
  No applicable method, applying
  {generic function do-emit-return (<object>, <object>, <object>, <object>) => (#rest)} to
  #[{<llvm-x86-darwin-back-end> #x019361A0}, {<llvm-module> #x019BB2D8}, t30,
    {<llvm-call-instruction> #x01CF0E40}].

emit c_callable_adder_c_wrapper_: Error:
  No applicable method, applying
  {generic function do-emit-return (<object>, <object>, <object>, <object>) => (#rest)} to
  #[{<llvm-x86-darwin-back-end> #x019361A0}, {<llvm-module> #x019BB2D8}, t30,
    {<llvm-call-instruction> #x01CF0E40}].
```

This is probably related to this code:

```dylan
define objc-selector @adder
  parameter target :: <NSObject>;
  parameter a :: <C-int>;
  result r :: <C-int>;
  selector: "adder:";
  type-encoding: "i@:i";
end;

define objc-method adder
    (target, selector, a :: <integer>)
 => (r :: <integer>)
  c-signature: (target :: <objc/instance>,
                selector :: <objc/selector>,
                a :: <C-int>) => (r :: <C-int>);
  assert-true(instance?(target, <test-class>));
  a + 1
end;

define objc-class <test-class> (<NSObject>) => DylanTestClass
  bind @adder => adder;
end;

define test objc-add-method-test ()
  let i = send($DylanTestClass, @alloc);
  assert-equal(3, send(i, @adder, 2));
end;
```

That ``objc-method`` definition is expanded with:

```dylan
define macro objc-method-definer
  { define objc-method ?:name (?args:*) => (?result:variable)
      c-signature: (?cffi-parameters:*) => (?cffi-result:variable);
      ?:body
    end }
    => {
         define function ?name (?args) => (?result)
           ?body
         end;
         define c-callable-wrapper ?name ## "-c-wrapper" of ?name
           ?cffi-parameters ;
           result ?cffi-result
         end;
  }

  { define objc-method ?:name (?args:*) => ()
      c-signature: (?cffi-parameters:*) => ();
      ?:body
    end }
    => {
         define function ?name (?args) => ()
           ?body
         end;
         define c-callable-wrapper ?name ## "-c-wrapper" of ?name
           ?cffi-parameters
         end;
  }

  args:
    { } => { }
    { ?arg:variable, ... } => { ?arg, ... }

  cffi-parameters:
    { } => { }
    { ?cffi-parameter:variable, ... } => { parameter ?cffi-parameter; ... }
end;
```

**Comment by waywardmonkeys** [comment]
@housel asked for DFM ...

The compiler message said it involved ``t36``, but I'm not sure exactly which of these blocks of code it is:

```dylan
METHOD adder-c-wrapper-initializer () => ()
  LOCAL METHOD (target :: <raw-c-pointer>, selector :: <raw-c-pointer>, a :: <raw-c-signed-int>) => (raw :: <raw-c-signed-int>)
          t1 := [PRIMOP cast-pointer-as-raw({{ target }})]
          t2 := [PRIMOP wrap-machine-word(t1)]
          Vt18 := [STACK-VECTOR (^#"address", t2)]
          t18 := [CONGRUENT-CALLg ^{<&generic> make}(^{<&class> <objc/instance>}, Vt18)]
          t5 := [CONGRUENT-CALLi ^{<&method> import-function-for-<objc/instance> (#P{ low-level-type-for(<objc/instance>) })}(t18)]
          t6 := [PRIMOP cast-pointer-as-raw({{ selector }})]
          t7 := [PRIMOP wrap-machine-word(t6)]
          Vt18 := [STACK-VECTOR (^#"address", t7)]
          t18 := [CONGRUENT-CALLg ^{<&generic> make}(^{<&class> <objc/selector>}, Vt18)]
          t29 := [PRIMOP wrap-abstract-integer({{ a }})]
          *t30(1) := [VALUES t29]
          *t19(1) := multiple-value-check-type *t30(1) :: ^{model-object :: <union>}
          t31 := *t19(1) [0]
          t13 := [CALLx ^{<&method> adder (<object>, <object>, <integer>)}(t5, t18, t31)]
          t36 := [PRIMOP unwrap-abstract-integer(t13)] // tail call
          return t36
        END
  t3 := [PRIMOP cast-pointer-as-raw(^(iep of)[C-CALLABLE-FUNCTION $s])]
  t4 := [PRIMOP wrap-machine-word(t3)]
  Vt7 := [STACK-VECTOR (^#"address", t4)]
  t7 := [CONGRUENT-CALLg ^{<&generic> make}(^{<&class> <c-function-pointer-instantiation>}, Vt7)]
  define {adder-c-wrapper in objc-test-suite} = t7
  *t8(0) := [VALUES]
  return *t8(0)
END
```

```dylan
METHOD (target :: <raw-c-pointer>, selector :: <raw-c-pointer>, a :: <raw-c-signed-int>) => (raw :: <raw-c-signed-int>)
  t1 := [PRIMOP cast-pointer-as-raw({{ target }})]
  t2 := [PRIMOP wrap-machine-word(t1)]
  Vt18 := [STACK-VECTOR (^#"address", t2)]
  t18 := [CONGRUENT-CALLg ^{<&generic> make}(^{<&class> <objc/instance>}, Vt18)]
  t5 := [CONGRUENT-CALLi ^{<&method> import-function-for-<objc/instance> (#P{ low-level-type-for(<objc/instance>) })}(t18)]
  t6 := [PRIMOP cast-pointer-as-raw({{ selector }})]
  t7 := [PRIMOP wrap-machine-word(t6)]
  Vt18 := [STACK-VECTOR (^#"address", t7)]
  t18 := [CONGRUENT-CALLg ^{<&generic> make}(^{<&class> <objc/selector>}, Vt18)]
  t29 := [PRIMOP wrap-abstract-integer({{ a }})]
  *t30(1) := [VALUES t29]
  *t19(1) := multiple-value-check-type *t30(1) :: ^{model-object :: <union>}
  t31 := *t19(1) [0]
  t13 := [CALLx ^{<&method> adder (<object>, <object>, <integer>)}(t5, t18, t31)]
  t36 := [PRIMOP unwrap-abstract-integer(t13)] // tail call
  return t36
END
```

**Comment by waywardmonkeys** [comment]
This was indeed fixed by efacb11. Thanks!
